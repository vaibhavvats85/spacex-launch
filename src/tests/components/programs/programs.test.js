import { shallow } from 'enzyme';
import Programs from '../../../components/programs/programs';
import React from 'react';

let wrapper;
jest.mock('react-redux', () => ({
    useDispatch: jest.fn(),
    useSelector: () => ({
        data: ['some data'],
        hasError: false,
        loading: false
    })
}));
jest.mock('react-router-dom', () => ({
    useLocation: () => jest.fn(),
    useHistory: () => jest.fn()
}));

beforeEach(() => {
    wrapper = shallow(<Programs />);
});

test('programs component should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should render cards if data is present', () => {
    const cards = wrapper.find('Cards');
    expect(cards.exists()).toBeTruthy();
});
