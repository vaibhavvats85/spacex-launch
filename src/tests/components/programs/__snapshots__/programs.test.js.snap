// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`programs component should render correctly 1`] = `
<div
  className="programs"
>
  <Cards
    lauchPrograms={
      Array [
        "some data",
      ]
    }
  />
</div>
`;
